server/schemas/schemaModel.js

```diff
@@ -28,32 +28,26 @@ module.exports = {
         //console.debug("findOneAsync args:", arguments);
         if (schema) {
           error = {message: "Schema already exists"};
-          helpers.errorHandler(error, request, response, next);
-          return;
+          next(error, false);
         } else {
-          Model.createAsync({
+          return Model.createAsync({
```

Comment:
createAsync is a result of require("bluebird").promisifyAll

It is a promise and I return it here so that any errors will bubble up into the initial 'catch' function. the success cases are handled by the callback.